autoload/ale/definition.vim

```diff
@@ -80,6 +80,47 @@ function! ale#definition#HandleLSPResponse(conn_id, response) abort
 
             call ale#definition#UpdateTagStack()
             call ale#util#Open(l:filename, l:line, l:column, l:options)
+
+            if l:filename[:5] is? 'jdt://'
+                let l:found = v:false
+
+                for l:linter in ale#linter#Get('java')
+                    if l:linter.name is# 'eclipselsp'
+                        let l:found = v:true
+                    endif
+                endfor
+
+                if ! l:found
+                    break
+                endif
+
+                " Display java file from jar library as part of current project.
+                set filetype=java
+
+                let b:ale_lsp_root = a:conn_id[stridx(a:conn_id, ':')+1:]
+                let s:cb_status = 'pending'
+
+                function! s:HandleClassFileContents(result) abort
+                    let s:cb_status = 'complete'
+                    let s:result = a:result
+                endfunction
+
+                call ale#lsp_linter#SendRequest(
+                \   bufnr(''),
+                \   'eclipselsp',
+                \   [0, 'java/classFileContents', {'uri': l:filename}],
+                \       function('s:handle_class_file_contents'))
+
+                while s:cb_status isnot# 'complete'
```

Comment:
I am not convinced blocking until the LSP response arrives is something we want to do.  Better have a callback like OnReady() that can take care of completing the task (e.g. opening a readonly buffer with the source on the desired position) after the call is completed.